CHECKOUT_mom6.csh

```diff
@@ -5,51 +5,19 @@ source $MODULESHOME/init/csh
 cd ../SHiELD_SRC/
 echo `pwd`
 
-
-
-# ---------------- component 'mom6'
-echo "Cloning https://github.com/NOAA-GFDL/ocean_BGC.git on branch/tag master"
-set git_output=`git clone -q --recursive -b master https://github.com/NOAA-GFDL/ocean_BGC.git >& /dev/stdout`
-if ( $? != 0 ) then
-     echo "$git_output" | sed 's/^/**GIT ERROR** /' > /dev/stderr
-     exit 1
+git clone -b dev/gfdl https://github.com/NOAA-GFDL/MOM6/
+git clone -b dev/gfdl https://github.com/NOAA-GFDL/SIS2/
+git clone -b dev/gfdl https://github.com/NOAA-GFDL/icebergs/
+
+(cd icebergs && git checkout dev/gfdl)
+if ("fffb6f35" != "") then
```

Comment:
Long-term I don't think it is a good idea to hard-code individual commit hashes. For this stage of development it is fine and probably necessary.

Reply:
Is it necessary to use this commit or does it work with dev/gfdl?  Is the motivation for using the static commit that we don't have CI setup?

Reply:
I haven't tried the latest dev/GFDL version, so I'm not sure if it will work right off the bat. If not, I don't anticipate many changes since this is not an old commit.
The static commit is taken from the original OMIP version we first started working on to get a prototype model up and running. It is probably a good idea to move to the most recent ocean version if you guys are comfortable with it.

Reply:
@breichl which MOM6 branch/tag/checkout do you recommend for us to use as a reasonably-stable version for coupled-model development? We had a similar question from Steve P at Sofar.

Reply:
One option if we wanted something stable and tested outside of GFDL is to use https://github.com/mom-ocean/MOM6.  This is probably the recommendation I would give to outside groups, it is updated less frequently and is tested by a consortium before merging updates.  But I would not prefer this approach for our development since it will not give us the ability to modify source code efficiently.

We could create our own fork and merge updates from dev/gfdl at semi-regular intervals.  The benefit is that we have control over what gets merged in and we can easily update source code.  But I don't think we need this level of control so this is probably not right for us either.

The more I think about it, what Joseph is doing is consistent with how we develop other models (e.g., OM4, OM5) where we point to a specific hash either in a .gitmodules file or in an XML/script and update the hash when we know there are changes to dev/GFDL that we want.  We can easily push code changes to dev/gfdl to update source code.  So I'm actually happy with this, we just need to remember to periodically update it.  Maybe we set these hashes near the top of the script as "MOM_VERSION" and "SIS_VERSION" to make it clear?

Reply:
Doesn't the dev/gfdl branch of MOM6 use tags to ensure people know what the stable releases are?  This idea of relying on hashes which comes with no overall description makes it hard to know at a glance where you are in the tree related to the top of the trunk.  At a minimum, is there a changelog or some other plain-language document describing the state of the hash?

Reply:
@breichl, Initially, I worked on forked branches of each component to get things up and running. Then, we merged all updates into our forks, and after some testing, we pushed our changes to the main repositories.

Now, we are using the latest code updates, as mentioned in the other CHECKOUT_code file. We could continue using the latest dev/gfdl branch to keep receiving all the updates. Once a significant implementation needs to be done and tested, the developer could work on their fork and then merge back (essentially similar to the process I initially described).

Reply:
> Doesn't the dev/gfdl branch of MOM6 use tags to ensure people know what the stable releases are? This idea of relying on hashes which comes with no overall description makes it hard to know at a glance where you are in the tree related to the top of the trunk. At a minimum, is there a changelog or some other plain-language document describing the state of the hash?

It may help to check w/ Alistair/Marshall/Bob to clarify, but here is how I understand it:

NOAA-GFDL/MOM6 hasn't really been traditionally tagged/released, except a few times before 2020 (e.g., for the Adcroft et al 2019 paper).  The "releases" are facilitated through the main branch at the mom-ocean/MOM6 level, since there is now a community of external groups also developing MOM6.  mom-ocean also doesn't use tags, but is only updated every couple of months when a member of the consortium pushes their dev branch.  I think the PR and commit summaries serve as the primary description of the change/release.

At GFDL I have seen date strings also added along with the hash in the XML, maybe that was to make it something a bit more interpretable but I do not know (I think only the hash actually determines the checkout, so the date strings seem sort of arbitrary otherwise).  Otherwise I think we also rely on the commit logs (which are enforced to be descriptive and include all details).



> @breichl, Initially, I worked on forked branches of each component to get things up and running. Then, we merged all updates into our forks, and after some testing, we pushed our changes to the main repositories.
> 
> Now, we are using the latest code updates, as mentioned in the other CHECKOUT_code file. We could continue using the latest dev/gfdl branch to keep receiving all the updates. Once a significant implementation needs to be done and tested, the developer could work on their fork and then merge back (essentially similar to the process I initially described).

We might be okay to point to dev/gfdl to stay up to date.  However, since the NOAA-GFDL MOM6 maintainers won't be regression testing our use cases they won't know if/when a change breaks our models.  Do we worry about this with any of the other components?

A separate but related issue is that I don't think MOM6 is routinely tested with the latest dev/GFDL FMS version (mom6-examples points to release 2022.02).  SHiELD uses the latest FMS and updates w/ each release?

Reply:
> A separate but related issue is that I don't think MOM6 is routinely tested with the latest dev/GFDL FMS version (mom6-examples points to release 2022.02). SHiELD uses the latest FMS and updates w/ each release?

It is because of this lack of MOM6 testing upstream dependencies, that the FMS dev team has integrated the MOM6 test suite into the automoated CI/CD system for FMS PRs.  See an example of it [here](https://github.com/NOAA-GFDL/FMS/actions/runs/10370070480/job/28707162037?pr=1572).